Formula/include-what-you-use.rb

```diff
@@ -0,0 +1,63 @@
+class IncludeWhatYouUse < Formula
+  desc "Tool to analyze #includes in C and C++ source files"
+  homepage "https://include-what-you-use.org/"
+  url "https://include-what-you-use.org/downloads/include-what-you-use-0.12.src.tar.gz"
+  sha256 "a5892fb0abccb820c394e4e245c00ef30fc94e4ae58a048b23f94047c0816025"
+
+  depends_on "cmake" => :build
+  depends_on "llvm" # include-what-you-use 0.12 is compatible with llvm 8.0
+
+  def install
+    # We do not want to symlink clang headers into HOMEBREW_PREFIX, so install
+    # to libexec to ensure that the resource path, which is always computed
+    # relative to the location of the include-what-you-use executable and is
+    # not configurable, is also located under libexec.
+    args = std_cmake_args + %W[
+      -DCMAKE_INSTALL_PREFIX=#{libexec}
+      -DCMAKE_PREFIX_PATH=#{Formula["llvm"].opt_lib}
+    ]
+
+    mkdir "build" do
+      system "cmake", *args, ".."
+      system "make"
+      system "make", "install"
+    end
+
+    bin.write_exec_script Dir["#{libexec}/bin/*"]
+
+    # include-what-you-use needs a copy of the clang headers to be located
+    # in a specific folder under its resource path.
+    mkdir_p libexec/"lib/clang/#{Formula["llvm"].version}/include"
+    cp_r Dir["#{Formula["llvm"].opt_lib}/clang/#{Formula["llvm"].version}/include/*"],
```

Comment:
This will need a revision bump every time there's a new llvm because of this, AFAICT.

Reply:
Any bottles should functionally be fine not being rebuilt for minor revisions of llvm, but otherwise, yes. Typically IWYU brings out a new release when there is a new major version of llvm, so forced revision bumps should be less than you might imagine.

Reply:
Is it OK for IWYU to be using llvm/clang headers from a *different*, *older* clang/llvm? (As long as it's the same major version.)

Because the llvm will get minor revisions, but the headers shipped by IWYU bottles will remain the same.

If this all seems kosher, then additional explanatory comments would be nice.

Reply:
Done. Also tweaked the test to ensure that it will fail if there is an issue with the headers.

Reply:
Awesome! You rock!